androidApp/src/main/java/com/pulselink/data/sms/SmsCloudSynchronizer.kt

```
@@ -0,0 +1,314 @@
+@Singleton
+class SmsCloudSynchronizer @Inject constructor(
+    @ApplicationContext private val context: Context,
+    private val smsRepository: SmsRepository,
+    private val auth: FirebaseAuth,
+    private val firestore: FirebaseFirestore,
+    private val settingsRepository: SettingsRepository,
+    private val deviceContactsRepository: DeviceContactsRepository
+) {
+    private val mutex = Mutex()
+
+    suspend fun sync(threadId: Long = -1L): Result<Unit> = mutex.withLock {
```

detekt (1.23.8)

[warning] The function sync appears to be too complex based on Cyclomatic Complexity (complexity: 37). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CyclomaticComplexMethod)